lib/navigation/preprocess_encoders.cpp

```diff
@@ -1,56 +1,106 @@
+#include "consts.hpp"
 #include "preprocess_encoders.hpp"
 
+#include "core/types.hpp"
+
 namespace hyped::navigation {
 
-EncodersPreprocessor::EncodersPreprocessor()
+EncodersPreprocessor::EncodersPreprocessor(core::ILogger &logger)
+    : logger_(logger),
+      num_consecutive_outliers_per_encoder_({0, 0, 0, 0}),
+      are_encoders_reliable_({true, true, true, true}),
+      num_reliable_encoders_(core::kNumEncoders),
+      max_consecutive_outliers_(10)
```

Comment:
this should be a `constexpr` instead of a `const` class member

Reply:
We initialized like this for the sake of consistency against other values but could do this alternatively if this is the preferred method?